sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -262,7 +262,7 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
               // This join could be very slow or OOM
               val buildSide = getSmallerSide(left, right)
               Seq(joins.BroadcastNestedLoopJoinExec(
-                planLater(left), planLater(right), buildSide, joinType, nonEquiCond))
+                planLater(left), planLater(right), buildSide, joinType, j.condition))
```

Comment:
I read the JIRA description. If the join condition is wrong, isn't this a bug?

Reply:
We would never go to this code path. For equi-join we always use smj by default.

Reply:
Isn't smj only for orderable joining keys?

Reply:
yes, smj only works with orderable key, but we have checked `orderable` at analyzer side(at `BinaryComparison.checkInputDataTypes`). If we go in the equi-join that means the key is orderable.

Reply:
Oh okay, so basically we cannot reach here if it is a equi-join with non-orderable keys. And by no means it runs into `createJoinWithoutHint` for equi-join for now.

This is only for preventing possible bug.